tests/e2e/api-core-tests/tests/products/products.test.js

```diff
@@ -0,0 +1,773 @@
+/**
+ * Internal dependencies
+ */
+const { createSampleData, deleteSampleData } = require( '../../data/products' );
+const { productsApi } = require('../../endpoints/products');
+
+/**
+ * Tests for the WooCommerce Products API.
+ *
+ * @group api
+ * @group products
+ *
+ */
+ describe( 'Products API tests', () => {
+
+	const PRODUCTS_COUNT = 20;
+	let sampleData;
+
+	beforeAll( async () => {
+		sampleData = await createSampleData();
+	}, 10000 );
```

Comment:
This was an arbitrary choice based on the tests timing out on my local machine. I assume it will run faster in CI, but I've had to bump this to `20000`+ depending on the machine I'm on..

Reply:
> I assume it will run faster in CI, but I've had to bump this to 20000+ depending on the machine I'm on..

Anecdotally, local environments vary somewhat but on my machine E2E tends to run up to 10% faster.